### PR TITLE
Upgrade to Beam 2.61.0

### PR DESCRIPTION
**Pull request by damccorm**

### CODE REVIEW

**Comment by damccorm** [comment]
Hm, I'm now seeing several errors in the dataplex template like:

```
[ERROR] testE2E_mainPathWithAllStepsEnabled(com.google.cloud.teleport.v2.templates.DataplexBigQueryToGcsTest)  Time elapsed: 1.408 s  <<< ERROR!
org.apache.beam.sdk.Pipeline$PipelineExecutionException: java.lang.IllegalStateException: getters require withGetterTarget.
	at org.apache.beam.runners.direct.DirectRunner$DirectPipelineResult.waitUntilFinish(DirectRunner.java:377)
	at org.apache.beam.runners.direct.DirectRunner$DirectPipelineResult.waitUntilFinish(DirectRunner.java:345)
	at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:218)
	at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:67)
	at org.apache.beam.sdk.Pipeline.run(Pipeline.java:325)
	at org.apache.beam.sdk.Pipeline.run(Pipeline.java:310)
	at com.google.cloud.teleport.v2.templates.DataplexBigQueryToGcsTest.runTransform(DataplexBigQueryToGcsTest.java:885)
	at com.google.cloud.teleport.v2.templates.DataplexBigQueryToGcsTest.runTransform(DataplexBigQueryToGcsTest.java:872)
	at com.google.cloud.teleport.v2.templates.DataplexBigQueryToGcsTest.testE2E_mainPathWithAllStepsEnabled(DataplexBigQueryToGcsTest.java:404)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.junit.JUnitSessionStore$1.evaluateSafely(JUnitSessionStore.java:55)
	at org.mockito.internal.junit.JUnitSessionStore$1.evaluate(JUnitSessionStore.java:43)
	at org.apache.beam.sdk.testing.TestPipeline$1.evaluate(TestPipeline.java:331)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:379)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:340)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:125)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:413)
Caused by: java.lang.IllegalStateException: getters require withGetterTarget.
	at org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.base.Preconditions.checkState(Preconditions.java:512)
	at org.apache.beam.sdk.values.Row$Builder.withFieldValueGetters(Row.java:843)
	at org.apache.beam.sdk.schemas.GetterBasedSchemaProvider$ToRowWithValueGetters.apply(GetterBasedSchemaProvider.java:135)
	at org.apache.beam.sdk.schemas.GetterBasedSchemaProvider$ToRowWithValueGetters.apply(GetterBasedSchemaProvider.java:114)
	at org.apache.beam.sdk.schemas.SchemaCoder.encode(SchemaCoder.java:121)
	at org.apache.beam.sdk.coders.Coder.encode(Coder.java:135)
	at org.apache.beam.sdk.coders.KvCoder.encode(KvCoder.java:73)
	at org.apache.beam.sdk.coders.KvCoder.encode(KvCoder.java:37)
	at org.apache.beam.sdk.util.CoderUtils.encodeToSafeStream(CoderUtils.java:86)
	at org.apache.beam.sdk.util.CoderUtils.encodeToByteArray(CoderUtils.java:70)
	at org.apache.beam.sdk.util.CoderUtils.encodeToByteArray(CoderUtils.java:55)
	at org.apache.beam.sdk.util.CoderUtils.clone(CoderUtils.java:168)
	at org.apache.beam.sdk.util.MutationDetectors$CodedValueMutationDetector.<init>(MutationDetectors.java:118)
	at org.apache.beam.sdk.util.MutationDetectors.forValueWithCoder(MutationDetectors.java:49)
	at org.apache.beam.runners.direct.ImmutabilityCheckingBundleFactory$ImmutabilityEnforcingBundle.add(ImmutabilityCheckingBundleFactory.java:115)
	at org.apache.beam.runners.direct.ParDoEvaluator$BundleOutputManager.output(ParDoEvaluator.java:305)
	at org.apache.beam.runners.core.SimpleDoFnRunner.outputWindowedValue(SimpleDoFnRunner.java:276)
	at org.apache.beam.runners.core.SimpleDoFnRunner.access$900(SimpleDoFnRunner.java:86)
	at org.apache.beam.runners.core.SimpleDoFnRunner$DoFnProcessContext.output(SimpleDoFnRunner.java:432)
	at org.apache.beam.sdk.transforms.DoFnOutputReceivers$WindowedContextOutputReceiver.output(DoFnOutputReceivers.java:89)
	at org.apache.beam.sdk.transforms.MapElements$2.processElement(MapElements.java:151)
```

This is very similar to https://github.com/apache/beam/pull/32081#issuecomment-2479607653 and its not clear to me why we didn't see this in release validation. cc/ @Abacn

**Comment by Abacn** [comment]
RC is fine. There is a stubborn GItHub Action cache at play here. During validation of RC2 I needed to SSH into the worker and force to delete maven repo right after "Setup Environment" run in the action.

The repo's GitHub Action worker is persistent GCE instance. It's local maven repository is also persistent, and gets loaded from github action cache every time the test run

**Comment by damccorm** [comment]
> RC is fine. There is a stubborn GItHub Action cache at play here. During validation of RC2 I needed to SSH into the worker and force to delete maven repo right after "Setup Environment" run in the action.
> 
> The repo's GitHub Action worker is persistent GCE instance. It's local maven repository is also persistent, and gets loaded from github action cache every time the test run

Hm ok - thanks. Yeah, I just validated that the published source is fine as well, so that is almost definitely the problem.

**Comment by Abacn** [comment]
The failed test is `CassandraToBigtableIT.testWritetimeReplication`,

see https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/2033/checks?check_run_id=33573087438

likely flakiness due to cassandra service

**Review by Abacn — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -32,6 +32,7 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <java.version>11</java.version>
+<!--    <os.detected.classifier>osx-x86_64</os.detected.classifier>-->
```
Comment (Abacn) [inline pom.xml:35]: this line was used to build on my M1 Mac, can be removed from the code going to be merged to main

Comment (damccorm) [reply pom.xml:35]: I actually liked leaving it, I think it is useful to have around for others building with an M1 (self included) - thoughts?

**Comment by damccorm** [comment]
> The failed test is `CassandraToBigtableIT.testWritetimeReplication`,
> 
> see https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/2033/checks?check_run_id=33573087438
> 
> likely flakiness due to cassandra service

For better or for worse, we actually need to fix these tests now (red blocks merging)

**Comment by damccorm** [comment]
> > The failed test is `CassandraToBigtableIT.testWritetimeReplication`,
> > see https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/2033/checks?check_run_id=33573087438
> > likely flakiness due to cassandra service
> 
> For better or for worse, we actually need to fix these tests now (red blocks merging), so I'll see if I can do that in a separate PR

Actually, looks like this self-healed, so I'm going to leave it for now

**Pull request merged**